services/src/main/java/org/keycloak/protocol/oidc/endpoints/LoginStatusIframeEndpoint.java

```diff
@@ -51,7 +55,20 @@ public LoginStatusIframeEndpoint(KeycloakSession session) {
     @GET
     @Produces(MediaType.TEXT_HTML_UTF_8)
     public Response getLoginStatusIframe(@QueryParam("version") String version) {
-        return returnIframeFromResources("login-status-iframe.html", version, session);
+        final var map = new HashMap<String, Object>();
+        final var isSecureContext = SecureContextResolver.isSecureContext(session);
+        final var adminBaseUri = session.getContext().getUri(UrlType.ADMIN).getBaseUri();
+        map.put("isSecureContext", isSecureContext);
+        map.put("resourceCommonUrl", Urls.themeRoot(adminBaseUri).getPath() + "/common/keycloak");
+
+        return IframeUtil.returnIframe(version, session, () -> {
+            try {
+                return FreemarkerUtils.loadTemplateFromClasspath(map, "login-status-iframe.ftl", getClass());
+            } catch (Exception e) {
+                logger.error("Failure when loading login-status-iframe.ftl", e);
+                return null;
```

Comment:
Would this not trigger a null pointer later on since `getResourceAsStream()` will be called with this `null` value? Perhaps it might just as well error out here? Not sure what is considered best practice in this case.

Reply:
No, it does not call `getResourceAsStream()` and it won't trigger `NullPointerException` . When this method returns null, it is handled here on this line https://github.com/keycloak/keycloak/pull/34551/files#diff-1bad86b3a44fb4631eddecd1d9f7ec38f93a6450a8161ad2babe8163ef59b2afR48 and you can see in the following code in `IframeUtil.returnIframe` that it will return `Response.Status.NOT_FOUND` (aka HTTP error 404). This is better than re-throwing the exception as that will cause returning `500`, which should not really be done. In case that file is not found, the `404` is better practice than `500` .

Reply:
Are you sure? It seems this lambda is [evaluated here](https://github.com/keycloak/keycloak/blob/462613bc328040bec87399698a3aa33e6d7fc431/services/src/main/java/org/keycloak/protocol/oidc/endpoints/IframeUtil.java#L33) where the result is passed directly into `IframeUtil.class.getResourceAsStream()`.

Reply:
Yes, it works like this:

When the `LoginStatusIframeEndpoint.getLoginStatusIframe` is called:
- The `LoginStatusIframeEndpoint.getLoginStatusIframe` invokes the `IframeUtil.returnIframe` (not the method `IframeUtil.returnIframeFromResources`)
- When the `LoginStatusIframeEndpoint.getLoginStatusIframe` calls the line `responseEntityProvider.get()`, it invokes the supplier, which was passed to it as an argument. This is the code in `LoginStatusIframeEndpoint` - [lines 65-70](https://github.com/keycloak/keycloak/pull/34551/files#diff-85c707c42bef5061b6a4833b7fdc84ae2083f35372338bbea1535ce9c65a7ffaR65-R69) .
- If this code returned null, it is evaluated on [this line](https://github.com/keycloak/keycloak/pull/34551/files#diff-1bad86b3a44fb4631eddecd1d9f7ec38f93a6450a8161ad2babe8163ef59b2afR48) to return 404
- Note there is not any interaction at all with the method `IframeUtil.returnIframeFromResources` and hence with the `IframeUtil.class.getResourceAsStream`)


When the `ThirdPartyCookiesIframeEndpoint.step1` (or `step2` is called)
- It invokes the `IframeUtil.returnIframeFromResources`
- When the `LoginStatusIframeEndpoint.getLoginStatusIframe` calls the line `responseEntityProvider.get()`, it invokes the supplier, which was passed to it as an argument. This is the code in `LoginStatusIframeEndpoint.getLoginStatusIframe`, which invokes the `IframeUtil.class.getResourceAsStream`
- In other words, this code for handling HTML files works exactly the same as before this PR. There is the supplier, which invokes the `IframeUtil.class.getResourceAsStream()`, which is exactly same behaviour as before.

Is it more clear now?

Reply:
Yes, thanks for the explainer!